### ISSUE TITLE
Image differencing: what to use as a reference image?

### ISSUE THREAD

**Issue by drphilmarshall** [DarkEnergyScienceCollaboration/Twinkles#247]
Hey @SimonKrughoff - what is the basic plan for the reference image to use in, say the DR10 Level 2 processing? @rbiswas4 guessed that it might be the DR9 coadd - at which point I wondered if we might use the DR10 coadd as an approximation to this? 

Another thing that came up in discussion was the possibility of doing multiple processings, to emulate working with the DR3 data, for example. If we did add this into our program, I guess we could use the DR3 coadd as the reference image for the DR3 image differencing?

I guess @wmwv might have some thoughts on this too. What would you like us to be using, Michael?

**Comment by drphilmarshall** [comment]
@SimonKrughoff @wmwv Did you see this question? Any idea what the DM diffimaging plan is?

**Comment by wmwv** [comment]
Sorry for the delay in response, what is the scope of the question:
1. What Twinkles should do?
2. What LSST will do in operation?

I think they have different answers for now.

1. Twinkles
Should use a seeing-matched co-add of all images with seeing better than some threshold.

2. LSST Operations
Needs to experiment with several different approaches to this.  There are several issues to consider:
  a. Seeing
  b. S/N
  c. Gaps
  d. Differential chromatic refraction

  2. a. Seeing
     One should use a template with a seeing better than the new image.  Or perhaps a template composed of images with very close to the same PSF.
  2. b. Combining 10-20 images is largely sufficient to reach the point where the contribution to the random photon noise from the template is a small fraction of the signal in the new image being subtracted.  Combining hundreds of images is unlikely to be a net gain, giving the additional complications of the other points in this section.
  2. c. There are gaps between CCDs, rafts, and bad pixels.  Dithering (both translational and rotational will address this).  There will be some calculating on ensuring a minimum covering fraction for each pixel (or perhaps for at least >90 % of pixels)
  2. d. Due to differential chromatic refraction, there will likely be significant gains by using templates drawn from a narrow range of airmass for a given field.  With an estimated SED model of existing sources, one can predict their chromatic distortion reasonably well, but it is still likely that airmass-dependent templates will be useful.

**Comment by SimonKrughoff** [comment]
Apologies from me as well.  I'll add to what @wmwv said.

For twinkles, I think it's straightforward to make a seeing matched coadd.  The question is how do we get a list of the appropriate images to coadd?  We don't keep any info about the visits as far as I know, so the only option is to go through every visit and compute some statistic on the PSF.  I can write a script to do that if the info is not available any other way.  Maybe @jchiang87 or @sethdigel knows if we have the FWHM for each visit stored someplace.

Regarding LSST, it is likely we will not have enough images in the templates to neglect the noise in the template.  It is made worse because the noise in the template is correlated after warping to the science image.  There are methods for de-correlating (see Zackay, Ofek, and Gal-Yamm 2016), but they are not easy to implement in real space.  We are looking into this sort of thing right now.

**Comment by jchiang87** [comment]
The psf FWHM is being stored in the CcdVisit table as the "seeing" column.  The value is being extracted from `calexp` via [this code](https://github.com/DarkEnergyScienceCollaboration/pserv/blob/master/python/desc/pserv/utils.py#L75)

**Comment by SimonKrughoff** [comment]
That's outstanding.  It may be a little hacky, but I think we can make a list of images to coadd from that database.

**Comment by drphilmarshall** [comment]
As I mentioned in #248, Zackay recommends we use the "Kaiser" algorithm when making coadds - he also suggested that we simply combine _all_ images, without selecting the best seeing ones. What do you think, @wmwv ? Seems like an easier thing to do!

**Comment by SimonKrughoff** [comment]
Zackay is here at the moment and his arguments are very compelling, but unfortunately a little hard to implement as it is a little difficult to deal with variable PSFs for technical reasons.  It would be fairly interesting to do the diffim with a Kaiser-like algorithm and with a clipped version.

**Comment by drphilmarshall** [comment]
Good idea. Can you add that to the diff im recipe please, so that both refs
get made when the recipe is pipelined? (Obvs we will edit the recipe a lot
between now and then but still.)
